saleor/graphql/warehouse/filters.py

```diff
@@ -0,0 +1,37 @@
+import django_filters
+from django.db.models import Q
+
+from ...warehouse.models import Warehouse
+from ..core.types import FilterInputObjectType
+
+
+def filter_string(qs, name, value):
+    lookup = "__".join([name, "icontains"])
+    return qs.filter(**{lookup: value})
+
+
+def filter_address(qs, _, value):
+    qs = qs.filter(
+        Q(address__street_address_1__icontains=value)
+        | Q(address__street_address_2__icontains=value)
+        | Q(address__city__icontains=value)
+        | Q(address__postal_code=value)
+        | Q(address__phone__contains=value)
+    )
+    return qs
+
+
+class WarehouseFilter(django_filters.FilterSet):
+    name = django_filters.CharFilter(method=filter_string)
```

Comment:
I would suggest adding some kind of slug or `external_id` field - warehouses will be most likely synced with externa;l systems so it would be nice to assign remote resource id.

Reply:
@krzysztofwolski I think this PR is huge enough and we can skip this part for now ;) Especially that we don't have any specific requirements about that.